source/api-documentation.txt

```diff
@@ -0,0 +1,14 @@
+=================
+API Documentation
+=================
+
+.. default-domain:: mongodb
+
+.. toctree::
+   :titlesonly:
+   :maxdepth: 1
+
+    Bson API <http://mongodb.github.io/mongo-java-driver/4.3/apidocs/bson/index.html>
+    Core Driver API <http://mongodb.github.io/mongo-java-driver/4.3/apidocs/mongodb-driver-core/index.html>
+    Sync Driver API <http://mongodb.github.io/mongo-java-driver/4.3/apidocs/mongodb-driver-sync/index.html>
```

Comment:
Would it be better to use the titles from the current Java docs?
https://mongodb.github.io/mongo-java-driver/4.2/apidocs/
E.g.
Bson
Core
Java Driver (modern API)
Java Driver (legacy API)
I think we also refer to the last one somewhere in the docs, so we might include that if so.